python/lsst/pipe/base/config.py

```diff
@@ -199,6 +203,54 @@ class to allow configuration of the connections class. This dynamically
         doc="Flag to enable/disable saving of log output for a task, enabled by default.",
     )
 
+    def applyConfigOverrides(
+        self,
+        instrument: Instrument | None,
+        taskDefaultName: str,
+        pipelineConfigs: Iterable[ConfigIR] | None,
+        parameters: ParametersIR,
+        label: str,
+    ) -> None:
+        r"""Apply config overrides to this config instance.
+
+        Parameters
+        ---------
+        instrument : `Instrument` or `None`
+            If this is not None, this is an `Instrument` instance which
+            will be used to apply instrumental overrides.
+        taskDefaultName : `str`
+            The default name associated with the `Task` class. This is
+            used with instrumental overrides.
+        pipelineConfigs : `Iterable` of `ConfigIR`
+            This is an iterable of `ConfigIR` objects that contain
+            overrides to apply to this config instance.
+        parameters : `ParametersIR`
+            Parameters defined in a Pipeline which are used in formatting
+            of config values across multiple `Task`\ s in a pipeline.
+        label : `str`
+            The label associated with this class's Task in a pipeline.
+        """
+        overrides = ConfigOverrides()
+        if instrument is not None:
+            overrides.addInstrumentOverride(instrument, taskDefaultName)
+        if pipelineConfigs is not None:
+            for subConfig in (configIr.formatted(parameters) for configIr in pipelineConfigs):
+                if subConfig.dataId is not None:
+                    raise NotImplementedError(
+                        "Specializing a config on a partial data id is not yet "
+                        "supported in Pipeline definition"
+                    )
+                # only apply override if it applies to everything
+                if subConfig.dataId is None:
+                    if subConfig.file:
+                        for configFile in subConfig.file:
+                            overrides.addFileOverride(os.path.expandvars(configFile))
+                    if subConfig.python is not None:
+                        overrides.addPythonOverride(subConfig.python)
+                    for key, value in subConfig.rest.items():
+                        overrides.addValueOverride(key, value)
```

Comment:
I'm guessing this is unchanged, but I would not have expected these overrides to be applied after the `python:` block when both exist, though I don't have a solid reason for that expectation.

Reply:
Value overrides are always last, as python block have the most side effects, importing things, retargeting, etc